extensions/vault/runtime/src/main/java/io/quarkus/vault/internal/client/VaultClientException.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.vault.internal.client;
+
+import io.quarkus.vault.VaultException;
+
+public class VaultClientException extends VaultException {
```

Comment:
Is it something that could get to the user? Wondering if it's internal or not?

Reply:
yes, it could. but I would not expect them to manipulate it directly. so I only added the base exception VaultException at io.quarkus.vault. what shall we do?
 - leave it as is
 - move exception subclasses into io.quarkus.vault (not my preferred choice)
 - create another internal exception that will be caught and wrapped into a VaultException?
what is your preference?